src/dg/dg.h

```diff
@@ -53,6 +53,27 @@ std::vector< real > project_function(
     const dealii::FESystem<dim,dim> &fe_output,
     const dealii::QGauss<dim> &projection_quadrature);
 
+/// FullMatrix using std::vector. It replaces dealii::FullMatrix to store AD variables.
+template<typename real2>
+class FullMatrix
+{
+    std::vector<std::vector<real2>> matrix;
```

Comment:
This is incredibly innefficient. The implementation of dealii::FullMatrix uses contiguous memory. Here you are effectively storing a vector of pointers to vectors.
yourFullMatrix [ address_of_vector1, address_of_vector2, address_of_vector3, ...]

dealii::FullMatrix = [ value11, value12, value13, value21, value22, value23, ...]
where [i,j] would access value [i*col + j].

Reply:
Hmm yes, I wrote a quick fix to get over compile errors. It's no longer required with dealii's release version and I have deleted it in the recent commit.